ios/StripeTerminalReactNative.swift

```diff
@@ -280,6 +280,39 @@ class StripeTerminalReactNative: RCTEventEmitter, DiscoveryDelegate, BluetoothRe
         }
     }
 
+    @objc(connectLocalMobileReader:resolver:rejecter:)
+    func connectLocalMobileReader(params: NSDictionary, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+        guard let reader = params["reader"] as? NSDictionary else {
+            resolve(Errors.createError(code: CommonErrorType.InvalidRequiredParameter, message: "You must provide a reader object"))
+            return
+        }
+
+        // since simulated readers don't contain `id` property we take serialNumber as a fallback
+        let readerId = reader["serialNumber"] as? String
+
+        guard let selectedReader = discoveredReadersList?.first(where: { $0.serialNumber == readerId }) else {
+            resolve(Errors.createError(code: CommonErrorType.InvalidRequiredParameter, message: "Could not find reader with id \(readerId ?? "")"))
+            return
+        }
+
+        let locationId = params["locationId"] as? String
+
+        let connectionConfig = LocalMobileConnectionConfiguration(
+            locationId: locationId ?? selectedReader.locationId ?? "",
+            merchantDisplayName: nil // use the location name,
```

Comment:
I already did this in a later commit https://github.com/mihaildu/stripe-terminal-react-native/commit/13c5737b94090bcb1e7e60a5bc25178921a2141a

While testing I kept adding commits on top of the main branch. It's probably easier if I open a PR straight to `stripe/stripe-terminal-react-native`. You can see all the changes here https://github.com/stripe/stripe-terminal-react-native/compare/main...mihaildu:stripe-terminal-react-native:main

My question is about delegating the local mobile reader to self. I noticed bluetooth reader does the same, but now for some reason I'm forced to implement the event handlers specific for local mobile reader, and not re-use the generic ones. So I have to implement 
```
func localMobileReader(_ reader: Reader, didStartInstallingUpdate update: ReaderSoftwareUpdate, cancelable: Cancelable?)
```
Instead of re-using
```
func reader(_ reader: Reader, didStartInstallingUpdate update: ReaderSoftwareUpdate, cancelable: Cancelable?) {
```
Any ideas?

Reply:
> It's probably easier if I open a PR straight to `stripe/stripe-terminal-react-native`.

Ah, I see. Ok, that sounds good. We can move these conversations there after it's open.

> My question is about delegating the local mobile reader to self. I noticed bluetooth reader does the same, but now for some reason I'm forced to implement the event handlers specific for local mobile reader, and not re-use the generic ones. So I have to implement
> 
> ```
> func localMobileReader(_ reader: Reader, didStartInstallingUpdate update: ReaderSoftwareUpdate, cancelable: Cancelable?)
> ```
> 
> Instead of re-using
> 
> ```
> func reader(_ reader: Reader, didStartInstallingUpdate update: ReaderSoftwareUpdate, cancelable: Cancelable?) {
> ```
> 
> Any ideas?

Unfortunately, I think we have to duplicate these methods until we combine the protocols in the native SDKs.